lib/ae_mdw/stats.ex

```diff
@@ -238,6 +238,25 @@ defmodule AeMdw.Stats do
     end
   end
 
+  @spec fetch_difficulty_stats(State.t(), pagination(), query(), range(), cursor()) ::
+          {:ok, {pagination_cursor(), [statistic()], pagination_cursor()}} | {:error, reason()}
+  def fetch_difficulty_stats(state, pagination, query, range, cursor) do
+    with {:ok, filters} <- Util.convert_params(query, &convert_blocks_param/1),
+         {:ok, {prev, counts, next}} <-
+           fetch_statistics(state, pagination, filters, range, cursor, {:blocks, :key}),
+         {:ok, {^prev, difficulties, ^next}} <-
+           fetch_statistics(state, pagination, filters, range, cursor, :difficulty) do
+      [difficulties, counts]
+      |> Enum.zip()
+      |> Enum.map(fn {%{count: difficulty, start_date: start_date, end_date: end_date},
+                      %{count: count, start_date: start_date, end_date: end_date}} ->
+        average = if count == 0, do: 0, else: Float.round(difficulty / count, 2)
```

Comment:
Difficulty is a pretty high number, decimals don't really matter. I would round it to the nearest value and return always an integer